modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,321 @@
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javafx.scene.robot;
+
+import static com.sun.javafx.FXPermissions.CREATE_ROBOT_PERMISSION;
+
+import java.util.Objects;
+
+import javafx.geometry.Point2D;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.GlassRobot;
+import com.sun.javafx.tk.Toolkit;
+
+/**
+ * A {@code Robot} is used for simulating user interaction such as
+ * typing keys on the keyboard and using the mouse as well as capturing
+ * graphical information without requiring a {@link javafx.scene.Scene}
+ * instance.
+ *
+ * @since 11
+ */
+public class Robot {
+
+    private final GlassRobot peer;
+
+    /**
+     * Constructs a new {@code Robot} that can be used for simulating user
+     * interactions.
+     */
+    public Robot() {
+        // Ensure we have proper permission for creating a robot.
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            sm.checkPermission(CREATE_ROBOT_PERMISSION);
+        }
+
+        peer = Toolkit.getToolkit().createRobot();
+        Objects.requireNonNull(peer);
+    }
+
+    /**
+     * Presses the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to press
+     */
+    public void keyPress(KeyCode keyCode) {
+        peer.keyPress(keyCode);
+    }
+
+    /**
+     * Releases the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to release
+     */
+    public void keyRelease(KeyCode keyCode) {
+        peer.keyRelease(keyCode);
+    }
+
+    /**
+     * Types the specified {@link KeyCode} key.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #keyPress(KeyCode)} followed by {@link #keyRelease(KeyCode)}.
+     * @param keyCode the key to type
+     */
+    public void keyType(KeyCode keyCode) {
+        keyPress(keyCode);
+        keyRelease(keyCode);
+    }
+
+    /**
+     * Returns the current mouse x-position in screen coordinates.
```

Comment:
Minor nit: we normally don't hyphenate x position anywhere else. I think `{@code x} position` would be better.